### PR TITLE
Support scale WSurfaceItem's size to allow avoid contents fuzzy

### PR DESCRIPTION
**Pull request by zccrs**
Add surfaceSizeRatio property to WSurfaceItem.
Add parentSurfaceItem property to WXWaylandSurfaceItem.

### CODE REVIEW

File: src/server/qtquick/wsurfaceitem.cpp [hunk 1]
```diff
@@ -530,6 +530,31 @@ void WSurfaceItem::setRightPadding(qreal newRightPadding)
...
+void WSurfaceItem::setSurfaceSizeRatio(qreal ssr)
+{
+    Q_D(WSurfaceItem);
+
+    if (qFuzzyCompare(ssr, d->surfaceSizeRatio))
+        return;
+    d->surfaceSizeRatio = ssr;
+    Q_EMIT surfaceSizeRatioChanged();
+
+    surfaceSizeRatioChange();
```
Comment (wineee) [inline src/server/qtquick/wsurfaceitem.cpp:548]:
```
Q_EMIT surfaceSizeRatioChanged();
 surfaceSizeRatioChange();
```
?

File: src/server/kernel/wsurface.cpp [hunk 2]
```diff
@@ -158,6 +166,26 @@ void WSurfacePrivate::updateBuffer()
...
 
+void WSurfacePrivate::updatePreferredBufferScale()
+{
+    if (explicitPreferredBufferScale > 0)
+        return;
+
+    float maxScale = 1.0;
+    for (auto o : outputs)
+        maxScale = std::max(o->scale(), maxScale);
+    preferredBufferScale = qCeil(maxScale);
```
Comment (wineee) [inline src/server/kernel/wsurface.cpp:177]: Why not qRound ？

Comment (zccrs) [reply src/server/kernel/wsurface.cpp:177]: Must ensure wl_surface's buffer scale greater than wl_output's scale.

File: src/server/qtquick/wsurfaceitem.cpp (hunk 1 above)
Comment (zccrs) [reply src/server/qtquick/wsurfaceitem.cpp:548]: `surfaceSizeRatioChanged` is a signal to notify this property changed.
`surfaceSizeRatioChange` is a virtual function to allow the derived class do someting when surface size ratio changed.

**Pull request merged**